app/src/main/java/com/chess/clock/fragments/TimeControlFragment.java

```diff
@@ -228,8 +216,16 @@ public void afterTextChanged(Editable s) {
                 }
             }
         });
-        setClockTextWatcher(secondsEt);
-        setClockTextWatcher(incrementSecondsEt);
+
+        actionCompletedCallback(nameEt, () -> {
+            String currentName = nameEt.getText().toString();
+            autoNamingEnabled = currentName.isEmpty() || currentName.equals(latestAutoName);
+        });
+        actionCompletedCallback(minutesEt, this::autoNaming);
+        actionCompletedCallback(incrementMinutesEt, this::autoNaming);
+        setClockTextWatcherWithCallback(secondsEt, this::autoNaming);
+        setClockTextWatcherWithCallback(incrementSecondsEt, this::autoNaming);
```

Comment:
Can we also call `autoNaming` on Save action or focus lost? I entered time and tapped save for new time control and it remained empty. The keyboard confirm may be a missed required step.

Reply:
Updated to be more interactive 👍